app/(dashboard)/enroll/[repositoryId]/leaderBoard.tsx

```diff
@@ -0,0 +1,190 @@
+"use client";
+
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import UserProfileSummary from "@/components/ui/userProfileSummary";
+import { useState } from "react";
+
+//assuming data from api is in decreasing order by points.
+const userProfiles = [
+  {
+    name: "Formbricks",
+    points: 50000,
+    avatarUrl: "",
+    userGitHubId: "dsfsdf",
+  },
+  {
+    name: "dasds | Javascript 🤝 React ",
+    points: 10000,
+    avatarUrl: "",
+    userGitHubId: "alice-wonderland",
+    showSettingButtons: false,
+  },
+  {
+    name: "Bob",
+    points: 8000,
+    avatarUrl: "",
+    userGitHubId: "bob-the-builder",
+  },
+  {
+    name: "Charlie",
+    points: 5000,
+    avatarUrl: "",
+    userGitHubId: "charlie-brown",
+    showSettingButtons: false,
+  },
+  {
+    name: "David",
+    points: 5500,
+    avatarUrl: "",
+    userGitHubId: "david-copperfield",
+  },
+  {
+    name: "Emily",
+    points: 4000,
+    avatarUrl: "",
+    userGitHubId: "emily-dickinson",
+    showSettingButtons: false,
+  },
+  {
+    name: "Frank",
+    points: 3500,
+    avatarUrl: "",
+    userGitHubId: "frankenstein",
+  },
+  {
+    name: "Grace",
+    points: 3000,
+    avatarUrl: "",
+    userGitHubId: "grace-hopper",
+    showSettingButtons: false,
+  },
+  {
+    name: "Henry",
+    points: 2500,
+    avatarUrl: "",
+    userGitHubId: "henry-ford",
+  },
+  {
+    name: "Isla",
+    points: 1500,
+    avatarUrl: "",
+    userGitHubId: "isla-fisher",
+    showSettingButtons: false,
+  },
+  {
+    name: "Jack",
+    points: 1100,
+    avatarUrl: "",
+    userGitHubId: "jack-london",
+  },
+  {
+    name: "Kim",
+    points: 800,
+    avatarUrl: "",
+    userGitHubId: "kim-kardashian",
+    showSettingButtons: false,
+  },
+  {
+    name: "Liam",
+    points: 500,
+    avatarUrl: "",
+    userGitHubId: "liam-neeson",
+  },
+  {
+    name: "Mia",
+    points: 400,
+    avatarUrl: "",
+    userGitHubId: "mia-farrow",
+    showSettingButtons: false,
+  },
+  {
+    name: "Noah",
+    points: 100,
+    avatarUrl: "",
+    userGitHubId: "noah-wyle",
+  },
+  {
+    name: "Olivia",
+    points: 50,
+    avatarUrl: "",
+    userGitHubId: "noah-wyldfge",
+  },
+];
+
+interface pointRangesInterface {
+  label: string;
+  value: string;
+  description: string;
+}
+export default function LeaderBoard() {
+  const [selectedPointsRange, setSelectedPointsRange] = useState<pointRangesInterface | null>(null);
```

Comment:
@manish-singh-bisht,  I recommend using query params for filters. This allows users to share filtered data through URLs, unlike useState.

Reply:
makes sense, sharing will become better. will commit this soon.

Reply:
while doing this, i realised that even if i do it and save state in url, but the default tab will always be "Project Detail" (check page.tsx file, app/(dashboard)/enroll/[repositoryId]/page.tsx).

i will have to change this file as well and save state for these three tabs leaderboard,projectdetail,issues ,so that we can jump directly to leaderboard. 

i dont want to make changes to page.tsx file without the permissions from the maintainers.

Reply:
Yes, I also realized this issue.

I propose 2 solutions - 

1. we can add active tab to query string.

2. We can add the following section to layout - app/(dashboard)/enroll/[repositoryId]/layout.tsx
![Screenshot 2024-03-24 093443](https://github.com/formbricks/oss.gg/assets/82360525/fc00a765-7fbe-48b1-8871-bd1c1a89ffe3)
And create a separate page for each tab.

@mattinannt, What are your thoughts on this?

Reply:
hey @Kritik-J - both of these make sense :) @manish-singh-bisht pls go ahead as proposed